packages/camera/android/src/main/java/dev/flutter/plugins/camera/CameraPlugin.java

```diff
@@ -0,0 +1,207 @@
+import io.flutter.plugin.common.MethodCall;
+import io.flutter.plugin.common.MethodChannel;
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
+import io.flutter.plugin.common.MethodChannel.Result;
+
+public class CameraPlugin implements FlutterPlugin, ActivityAware, MethodCallHandler {
+
+  private final CameraPermissions cameraPermissions = new CameraPermissions();
+  private EventChannel imageStreamChannel;
+  private Camera camera;
+
+  private FlutterPluginBinding pluginBinding;
+  private ActivityPluginBinding activityBinding;
+
+  @Override
+  public void onAttachedToEngine(FlutterPluginBinding flutterPluginBinding) {
+    this.pluginBinding = flutterPluginBinding;
+    this.imageStreamChannel = new EventChannel(
+        this.pluginBinding.getFlutterEngine().getDartExecutor(),
+        "plugins.flutter.io/camera/imageStream"
+    );
+  }
+
+  @Override
+  public void onDetachedFromEngine(FlutterPluginBinding flutterPluginBinding) {
+    this.pluginBinding = null;
+  }
+
+  @Override
+  public void onAttachedToActivity(ActivityPluginBinding activityPluginBinding) {
+    // Setup
+    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
+      // When a background flutter view tries to register the plugin, the registrar has no activity.
+      // We stop the registration process as this plugin is foreground only. Also, if the sdk is
+      // less than 21 (min sdk for Camera2) we don't register the plugin.
+      return;
+    }
+
+    this.activityBinding = activityPluginBinding;
+
+    final MethodChannel channel =
+        new MethodChannel(pluginBinding.getFlutterEngine().getDartExecutor(), "plugins.flutter.io/camera");
+
+    channel.setMethodCallHandler(this);
+  }
+
+  @Override
+  public void onDetachedFromActivityForConfigChanges() {
+    // Ignore
+  }
+
+  @Override
+  public void onReattachedToActivityForConfigChanges(ActivityPluginBinding activityPluginBinding) {
+    // Ignore
+  }
+
+  @Override
+  public void onDetachedFromActivity() {
+    // Teardown
+    this.activityBinding = null;
+  }
+
+  private void instantiateCamera(MethodCall call, Result result) throws CameraAccessException {
+    String cameraName = call.argument("cameraName");
+    String resolutionPreset = call.argument("resolutionPreset");
+    boolean enableAudio = call.argument("enableAudio");
+    camera = new Camera(
+        activityBinding.getActivity(),
+        pluginBinding.getFlutterEngine().getRenderer(),
+        cameraName,
+        resolutionPreset,
+        enableAudio
+    );
+
+    EventChannel cameraEventChannel =
+        new EventChannel(
+            pluginBinding.getFlutterEngine().getDartExecutor(),
+            "flutter.io/cameraPlugin/cameraEvents" + camera.getFlutterTexture().id());
+    camera.setupCameraEventChannel(cameraEventChannel);
+
+    camera.open(result);
+  }
+
+  @Override
+  public void onMethodCall(@NonNull MethodCall call, @NonNull final Result result) {
```

Comment:
Can we find a way to not duplicate all this code?

Reply:
Sure, but it looks like this PR probably won't merge because @bparrishMines has other plans for the plugin. I would recommend that each plugin dev do whatever is most appropriate in terms of replication as they migrate.

Reply:
Would be useful to have a sample PR/commit that shows only the diff that's relevant to properly extending a plugin to the new embedder.

The camera refactoring plans are longer term, we will want to extend the existing plugin to support the new embedder much sooner, and I won't tie doing that to the big refactoring planned.

Reply:
Ok. I'll put up a PR that does the minimum possible work and shares code.